azure/endpoint/dns.tf

```diff
@@ -0,0 +1,30 @@
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip#attributes-reference
+# Public IP Addresses aren't allocated until they're attached to a device (e.g. a Virtual Machine/Load Balancer).
+# Instead you can obtain the IP Address once the Public IP has been assigned via the azurerm_public_ip Data Source.
+data "azurerm_public_ip" "main" {
+  name                = azurerm_public_ip.main.name
+  resource_group_name = data.azurerm_resource_group.main.name
+}
+
+data "azurerm_dns_zone" "main" {
+  name                = var.dns_zone_name
+  resource_group_name = var.dns_zone_resource_group_name
+}
+
+resource "azurerm_dns_a_record" "main" {
+  name                = var.subdomain
+  zone_name           = data.azurerm_dns_zone.main.name
+  resource_group_name = data.azurerm_dns_zone.main.resource_group_name
+  ttl                 = 300
+  target_resource_id  = azurerm_public_ip.main.id
+  # records             = [data.azurerm_public_ip.main.ip_address]
+}
+
+resource "azurerm_dns_cname_record" "main" {
+  name                = "asuid.${var.subdomain}"
+  zone_name           = data.azurerm_dns_zone.main.name
+  resource_group_name = data.azurerm_dns_zone.main.resource_group_name
+  ttl                 = "300"
+  record              = "${var.subdomain}.westus.cloudapp.azure.com"
```

Comment:
The region of the DNS record changes with the region that the VM is located in: 
![image](https://user-images.githubusercontent.com/8037512/214649876-fbd5857f-a599-463a-9ebf-871df0eb6926.png)

If that's intentional and still should be set to `westus`, LMK and I'll approve it.

Reply:
Ahh great catch, thank you! 👏 👏 . I _do_ need to pass in the region here.